src/ConnectionRequest.jl

```diff
@@ -23,18 +23,47 @@ abstract type ConnectionPoolLayer{Next <: Layer} <: Layer end
 export ConnectionPoolLayer
 
 function request(::Type{ConnectionPoolLayer{Next}}, url::URI, req, body;
+                 proxy=nothing,
                  socket_type::Type=TCPSocket, kw...) where Next
 
+    if proxy != nothing
+        target_url = url
+        url = URI(proxy)
+        if target_url.scheme == "http"
+            req.target = string(target_url)
+        end
+    end
+
     IOType = ConnectionPool.Transaction{sockettype(url, socket_type)}
     io = getconnection(IOType, url.host, url.port; kw...)
 
     try
+
+        if proxy != nothing && target_url.scheme == "https"
+            target = "$(URIs.hoststring(target_url.host)):$(target_url.port)"
+            @debug 1 "📡  CONNECT HTTPS tunnel to $target"
+            r = Request("CONNECT", target)
+            writeheaders(io, r)
+            startread(io)
+            readheaders(io, r.response)
+            if r.response.status != 200
+                return r
+            end
+            io = ConnectionPool.sslupgrade(io, target_url.host; kw...)
+        end
```

Comment:
To proxy HTTPS, we set the target to "host:port" and send a `CONNECT` request to establish a tunnel to the target. If that fails, just return the non-200 Response object. It will be converted into an exception by the ExceptionLayer, retried, etc as usual. If we get a 200 response, call `sslupgrade` to change the `Transaction{TCPSocket}` into a `Transaction{SSLContext}`.